gossip/client.go

```diff
@@ -147,7 +147,7 @@ func (c *client) handleGossip(g *Gossip, call *netrpc.Call) error {
 				log.Infof("received %d (%d fresh) info(s) from %s", infoCount, freshCount, c.addr)
 			}
 		}
-	} else {
+	} else if reply.Alternate != nil {
```

Comment:
The relationship between this log line and `reply.Alternate` being non-nil is unclear.